--- tmp_orig/nemo-6.6.3+zena/libnemo-private/nemo-progress-info.c	2026-01-08 14:29:12.000000000 +0100
+++ build/nemo-6.6.3+zena/libnemo-private/nemo-progress-info.c	2026-07-17 08:35:15.252049923 +0200
@@ -70,6 +70,10 @@
 	gboolean changed_at_idle;
 	gboolean progress_at_idle;
     gboolean queue_at_idle;
+
+	/* bandwidth graph */
+	gdouble transfer_rate;
+	gboolean nemo_delete;
 };
 
 struct _NemoProgressInfoClass
@@ -683,4 +687,42 @@
     g_mutex_unlock (&info->info_lock);
 
     return elapsed;
+}
+
+void
+nemo_progress_info_set_speed (NemoProgressInfo *info,
+                               gdouble           bytes_per_sec)
+{
+    g_mutex_lock (&info->info_lock);
+    info->transfer_rate = bytes_per_sec;
+    g_mutex_unlock (&info->info_lock);
+}
+
+gdouble
+nemo_progress_info_get_speed (NemoProgressInfo *info)
+{
+    gdouble ret;
+    g_mutex_lock (&info->info_lock);
+    ret = info->transfer_rate;
+    g_mutex_unlock (&info->info_lock);
+    return ret;
+}
+
+void
+nemo_progress_info_set_delete_mode (NemoProgressInfo *info,
+                                     gboolean          is_delete)
+{
+    g_mutex_lock (&info->info_lock);
+    info->nemo_delete = is_delete;
+    g_mutex_unlock (&info->info_lock);
+}
+
+gboolean
+nemo_progress_info_get_delete_mode (NemoProgressInfo *info)
+{
+    gboolean ret;
+    g_mutex_lock (&info->info_lock);
+    ret = info->nemo_delete;
+    g_mutex_unlock (&info->info_lock);
+    return ret;
 }
\ Kein Zeilenumbruch am Dateiende.
